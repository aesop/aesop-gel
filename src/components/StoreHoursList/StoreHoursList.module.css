@import 'styles/index';

.base {
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(15px);

  &,
  & * {
    box-sizing: border-box;
  }
}

.heading {
  margin-bottom: 6px;
}

.list {
  padding: 0;
  margin: 0;
  margin-bottom: 15px;
  line-height: 1.4;
  list-style-type: none;
}

.item {
  padding: 0;
  margin: 0;
  margin-bottom: 4px;
}

.dayName {
  font-family: var(--font-family-sans-serif-medium);
  font-size: rem(14px);

  &::after {
    content: ': ';
  }
}

.hours {
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(13px);
}

.alternateHours {
  color: var(--color-adjustment);

  &::after {
    content: '*';
  }
}

.alternateHoursNote {
  margin-bottom: 15px;
  color: var(--color-adjustment);
  font-size: rem(13px);

  &::before {
    content: '*';
  }
}
