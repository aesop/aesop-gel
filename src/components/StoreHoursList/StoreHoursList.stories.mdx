import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import StoreHoursList from './StoreHoursList';
import StoreHoursListFixture from './StoreHoursList.fixture';

<Meta title="Components/StoreHoursList" component={StoreHoursList} />

# StoreHoursList

<Preview>
  <Story name="Base component">
    <StoreHoursList
      alternateHoursNote={StoreHoursListFixture.alternateHoursNote}
      heading={StoreHoursListFixture.heading}
      hoursList={StoreHoursListFixture.hoursList}
    />
  </Story>
</Preview>

## API

<Props of={StoreHoursList} />
