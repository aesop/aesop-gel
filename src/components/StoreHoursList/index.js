/* istanbul ignore file */
import { StoreHoursList } from './StoreHoursList';

export { StoreHoursList };
