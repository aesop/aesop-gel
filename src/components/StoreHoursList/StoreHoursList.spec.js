import React from 'react';
import { render } from '@testing-library/react';
import StoreHoursList from './StoreHoursList';
import StoreHoursListFixture from './StoreHoursList.fixture';

describe('<StoreHoursList />', () => {
  it('should be defined', () => {
    expect(StoreHoursList).toBeDefined();
  });

  it('should render base component correctly', () => {
    const { container } = render(
      <StoreHoursList
        alternateHoursNote={StoreHoursListFixture.alternateHoursNote}
        heading={StoreHoursListFixture.heading}
        hoursList={StoreHoursListFixture.hoursList}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
