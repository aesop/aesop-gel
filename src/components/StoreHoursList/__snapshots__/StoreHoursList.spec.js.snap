// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoreHoursList /> should render base component correctly 1`] = `
<div>
  <div
    class="base dark"
  >
    <h4
      class="base xXSmall dark heading"
    >
      Opening hours
    </h4>
    <ul
      class="list"
    >
      <li
        class="item"
      >
        <span
          class="dayName"
        >
          Monday
        </span>
        <span
          class="hours"
        >
          11am - 4pm
        </span>
      </li>
      <li
        class="item"
      >
        <span
          class="dayName"
        >
          Tuesday
        </span>
        <span
          class="hours alternateHours"
        >
          10am - 3pm
        </span>
      </li>
      <li
        class="item"
      >
        <span
          class="dayName"
        >
          Wednesday
        </span>
        <span
          class="hours"
        >
          10am - 3pm
        </span>
      </li>
      <li
        class="item"
      >
        <span
          class="dayName"
        >
          Thursday
        </span>
        <span
          class="hours"
        >
          10am - 3pm
        </span>
      </li>
      <li
        class="item"
      >
        <span
          class="dayName"
        >
          Friday
        </span>
        <span
          class="hours"
        >
          10am - 3pm
        </span>
      </li>
      <li
        class="item"
      >
        <span
          class="dayName"
        >
          Saturday
        </span>
        <span
          class="hours alternateHours"
        >
          10am - 3pm
        </span>
      </li>
      <li
        class="item"
      >
        <span
          class="dayName"
        >
          Sunday
        </span>
        <span
          class="hours"
        >
          10am - 3pm
        </span>
      </li>
    </ul>
    <div
      class="alternateHoursNote"
    >
      Special opening hours
    </div>
  </div>
</div>
`;
